content/CustomExceptions/CustomExceptions.q.md

```diff
@@ -0,0 +1,1501 @@
+
+ - syntax error, unexpected `'Exception'` (T_STRING), expecting `'{'` in /DivideByZeroException.php on line number 2
+
+
+/// type=MS, answer=[4,5]
+
+Which statements correctly describe the error?
+
+ - There is no `new` keyword specified after `throw` on line 7 of `Main.php`.
+
+ - There is no semicolon `;` at the end of the statement on line 7 of `Main.php`.
+
+ - There is no open curly brace `{` after the `if` construct on line 6 of `Main.php`.
+
+ - In the `throw` statement on line 7 of `Main.php`, the `throw` keyword is misspelled as `throws`.
+
+ - On line 7 of `Main.php`, the statement `throws new DivideByZeroException("The value '0' as a divisor is not allowed.");` is invalid.
+
+:::
+
+
+/// type=CR, answer=[tests/CustomExceptions/MisspelledThrowKeywordTest.php], filename=[Main.php,DivideByZeroException.php]
+
+Correct the code so that it outputs the string `Exception caught: The value '0' as a divisor is not allowed.`.
+
+```php
+<?php
+require_once(__DIR__ . "/DivideByZeroException.php");
+
+function divide($dividend, $divisor)
+{
+    if ($divisor == 0) {
+        throws new DivideByZeroException("The value '0' as a divisor is not allowed.");
+    } else {
+        return ($dividend / $divisor);
+    }
+}
+
+try {
+    echo divide(10, 0);
+} catch (DivideByZeroException $div) {
+    echo "Exception caught: " . $div->errorMessage();
+}
+?>
+```
+
+```php
+<?php
+class DivideByZeroException extends Exception
+{
+    public function errorMessage()
+    {
+        return $this->getMessage();
+    }
+}
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[Main.php,DivideByZeroException.php,DivideByNegativeException.php]
+
+```php
+<?php
+require_once(__DIR__ . "/DivideByZeroException.php");
+
+function divide($dividend, $divisor)
+{
+    if ($divisor == 0) {
+        throw new DivideByZeroException("The value '0' as a divisor is not allowed.");
+    } elseif ($divisor < 0) {
+        throw new DivideByNegativeException();
+    } else {
+        return ($dividend / $divisor);
+    }
+}
+
+try {
+    echo divide(10, -2);
+} catch (DivideByZeroException $div) {
+    echo "Exception caught: " . $div->errorMessage();
+} catch (DivideByNegativeException $neg) {
+    echo "Exception caught: " . $neg->myErrorMessage() . " in " . $neg->myErrorFile();
+}
+?>
+```
+
+```php
+<?php
+class DivideByZeroException extends Exception
+{
+    public function errorMessage()
+    {
+        return $this->getMessage();
+    }
+}
+?>
+```
+
+```php
+<?php
+class DivideByNegativeException extends Exception 
+{
+    public function myErrorMessage()
+    {
+        return "A negative number as a divisor is encountered";
+    }
+
+    public function myErrorFile()
+    {
+        return $this->getFile();
+    }
+}
+?>
+```
+/// type=MS, answer=[2,3]
+
+Execute the program. What are the error messages?
+
+ - Undefined variable: `new` in /Main.php on line number 7
+
+ - Error: Class `'DivideByNegativeException'` not found in /Main.php on line number 9
```

Comment:
This is the error messages displayed in my local. Right now, our reader cannot capable to test multiple files yet.
![image](https://user-images.githubusercontent.com/25945790/76576143-36136b80-64fc-11ea-8c26-7a334cb0c005.png)

By the way, in the options provided I only use `/Main.php` instead of the complete path shown in the image since this is how our reader generate it using single file. Please check. Thank you.